frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -47,35 +50,23 @@ interface PublicFormProviderProps {
   children: React.ReactNode
 }
 
-export const PublicFormProvider = ({
-  formId,
-  children,
-}: PublicFormProviderProps): JSX.Element => {
-  // Once form has been submitted, submission data will be set here.
-  const [submissionData, setSubmissionData] = useState<SubmissionData>()
+export function useCommonFormProvider(
+  formId: string,
+  data: PreviewFormViewDto | PublicFormViewDto | undefined,
+  error: ApiError | null,
```

Comment:
suggest not passing in any args to this hook, it should only contain the truly shared variables imho. Other implementations can probably be duplicated.

Reply:
makes sense, i've removed `data` and `error` but have left `formId` in